### PR TITLE
pid1: refactoring of unit state machine logging and unit timer refactoring

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
just some housekeeping. Nothing new. No real code changes.

File: src/core/service.c [hunk 1]
```diff
@@ -2679,23 +2670,19 @@ static void service_run_next_control(Service *s) {
...
-                service_enter_running(s, SERVICE_SUCCESS);
-        } else
-                service_enter_stop(s, SERVICE_FAILURE_RESOURCES);
+        if (r < 0) {
+                log_unit_warning_errno(UNIT(s), r, "Failed to spawn next control task: %m");
+
+                if (IN_SET(s->state, SERVICE_CONDITION, SERVICE_START_PRE, SERVICE_START_POST, SERVICE_STOP))
+                        service_enter_signal(s, SERVICE_STOP_SIGTERM, SERVICE_FAILURE_RESOURCES);
+                else if (s->state == SERVICE_STOP_POST)
+                        service_enter_dead(s, SERVICE_FAILURE_RESOURCES, true);
```
Comment (yuwata) [inline src/core/service.c]: Please add a short comment for the boolean flag.

File: src/core/socket.c [hunk 2]
```diff
@@ -515,22 +515,19 @@ int socket_acquire_peer(Socket *s, int fd, SocketPeer **p) {
 
         assert(fd >= 0);
         assert(s);
+        assert(ret);
 
         if (getpeername(fd, &sa.peer.sa, &sa.peer_salen) < 0)
-                return log_unit_error_errno(UNIT(s), errno, "getpeername failed: %m");
+                return log_unit_error_errno(UNIT(s), errno, "getpeername failed(): %m");
```
Comment (yuwata) [inline src/core/socket.c]: Hm? Inserted `()` at wrong place?

File: src/core/socket.c [hunk 3]
```diff
@@ -2422,21 +2413,19 @@ static void socket_enter_running(Socket *s, int cfd_in) {
...
                         service_release_socket_fd(SERVICE(service));
-                        goto fail;
+                        goto queue_error;
                 }
 
                 /* Notify clients about changed counters */
                 unit_add_to_dbus_queue(UNIT(s));
         }
-
-        TAKE_FD(cfd);
```
Comment (yuwata) [inline src/core/socket.c:2435]: The change is not trivial for me. Could you explain about the change?
Maybe, it is better to split out to another commit or mention the change in the commit message.

File: src/core/mount.c [hunk 4]
```diff
@@ -1163,23 +1171,27 @@ static void mount_enter_mounting(Mount *m) {
                         r = exec_command_append(m->control_command, "-t", p->fstype, NULL);
                 if (r >= 0 && !isempty(opts))
                         r = exec_command_append(m->control_command, "-o", opts, NULL);
-        } else
-                r = -ENOENT;
-        if (r < 0)
+                if (r < 0) {
+                        log_unit_warning_errno(UNIT(m), r, "Failed to prepare mount command line: %m");
+                        goto fail;
+                }
```
Comment (yuwata) [inline src/core/mount.c]: I'd like to move the error handling just after `exec_command_set()`. Then we can drop multiple `r >= 0` conditions.

File: src/core/mount.c [hunk 5]
```diff
@@ -1223,23 +1235,28 @@ static void mount_enter_remounting(Mount *m) {
                         r = exec_command_append(m->control_command, "-w", NULL);
                 if (r >= 0 && p->fstype)
                         r = exec_command_append(m->control_command, "-t", p->fstype, NULL);
-        } else
-                r = -ENOENT;
-        if (r < 0)
+                if (r < 0) {
```
Comment (yuwata) [inline src/core/mount.c:1262]: Also here.

File: src/core/automount.c [hunk 6]
```diff
@@ -355,8 +356,9 @@ static int open_dev_autofs(Manager *m) {
 
         init_autofs_dev_ioctl(&param);
         if (ioctl(m->dev_autofs_fd, AUTOFS_DEV_IOCTL_VERSION, &param) < 0) {
+                r = -errno;
```
Comment (yuwata) [inline src/core/automount.c]: `RET_NERRNO()` ?

File: src/core/timer.c [hunk 7]
```diff
@@ -539,37 +547,44 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
...
+                if (r < 0) {
+                        log_unit_warning_errno(UNIT(t), r, "Failed to disable realtime event source: %m");
                         goto fail;
+                }
         }
 
         timer_set_state(t, TIMER_WAITING);
         return;
 
 fail:
```
Comment (yuwata) [inline src/core/timer.c:586]: Not a matter, but the error handling at the beginning of the function can also use `goto fail;`.

File: src/core/timer.c [hunk 8]
```diff
@@ -615,7 +632,6 @@ static void timer_enter_running(Timer *t) {
...
```
Comment (yuwata) [inline src/core/timer.c:632]: Also here. we can use `goto fail` on vanished case.

**Review by yuwata — CHANGES_REQUESTED** [review]
One question, and several minor requests.

**Comment by poettering** [comment]
Rebased and addressed all raised issues.

**Review by bluca — APPROVED** [review]

**Pull request merged**